### PR TITLE
Add notebook on Authenticating to Vespa Cloud

### PR DESCRIPTION
**Pull request by thomasht86**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

Added a notebook to demonstrate and explain the different authentication methods, as this is something that is often a source of misunderstanding/misconfiguration, ref #862.

Please have a look at the table in the notebook to see if I missed something. 
Adding below for convenience: 
<details>

For background context, it is useful to read the [Vespa Cloud Security Guide](https://cloud.vespa.ai/en/security/guide).

## Control-plane vs Data-plane

This may be self-explanatory for some, but it is worth mentioning that Vespa Cloud has two main components: the control-plane and the data-plane, which provide access to different functionalities.

|                                                                                              | Control-plane | Data-plane | Comments                                                                                   |
| -------------------------------------------------------------------------------------------- | ------------- | ---------- | ------------------------------------------------------------------------------------------ |
| Deploy application                                                                           | ✅            | ❌         |                                                                                            |
| Modify application (re-deploy)                                                               | ✅            | ❌         |                                                                                            |
| Add or modify data-plane certs or token(s)                                                   | ✅            | ❌         |                                                                                            |
| Feed data                                                                                    | ❌            | ✅         |                                                                                            |
| Query data                                                                                   | ❌            | ✅         |                                                                                            |
| Delete data                                                                                  | ❌            | ✅         |                                                                                            |
| [Visiting](https://docs.vespa.ai/en/visiting.html)                                           | ❌            | ✅         |                                                                                            |
| [Monitoring](https://cloud.vespa.ai/en/monitoring)                                           | ❌            | ✅         |                                                                                            |
| Get application package                                                                      | ✅            | ❌         |                                                                                            |
| [`vespa auth login`](https://docs.vespa.ai/en/reference/vespa-cli/vespa_auth_login.html)     | ✅            | ❌         | Interactive control-plane login in browser                                                 |
| [`vespa auth api-key`](https://docs.vespa.ai/en/reference/vespa-cli/vespa_auth_api-key.html) | ✅            | ❌         | Headless control-plane authentication with an API key generated in the Vespa Cloud console |
| [`vespa auth cert`](https://docs.vespa.ai/en/reference/vespa-cli/vespa_auth_cert.html)       | ❌            | ✅         | Used to generate a certificate for a data-plane connection                                 |
| [`VespaCloud`](https://pyvespa.readthedocs.io/en/latest/reference-api.html#vespacloud)       | ✅            | ❌         | `VespaCloud` is a control-plane connection to Vespa Cloud                                  |
| [`VespaDocker`](https://pyvespa.readthedocs.io/en/latest/reference-api.html#vespadocker)     | ✅            | ❌         | `VespaDocker` is a control-plane connection to a Vespa server running in Docker            |
| [`Vespa`](https://pyvespa.readthedocs.io/en/latest/reference-api.html#vespa)                 | ❌            | ✅         | `Vespa` is a data-plane connection to an existing Vespa application                        |

</details>

I also discovered that a lot of the recently added classes had not been added to the reference API-documentation, so included these.

### CODE REVIEW

File: docs/sphinx/source/authenticating-to-vespa-cloud.ipynb [hunk 1]
```diff
@@ -0,0 +1,984 @@
...
+    "  <source media=\"(prefers-color-scheme: light)\" srcset=\"https://vespa.ai/assets/vespa-ai-logo-rock.svg\">\n",
+    "  <img alt=\"#Vespa\" width=\"200\" src=\"https://vespa.ai/assets/vespa-ai-logo-rock.svg\" style=\"margin-bottom: 25px;\">\n",
+    "</picture>\n",
+    "\n",
+    "# Authenticating to Vespa Cloud\n",
+    "\n",
+    "Security is a top priority for the Vespa Team.\n",
+    "We understand that as a newcomer to Vespa, the different authentication methods may not always be immediately clear.\n",
+    "\n",
+    "This notebook is intended to provide some clarity on the different authentication methods needing when interacting with Vespa Cloud for different purposes.\n"
```
Comment (kkraune) [inline docs/sphinx/source/authenticating-to-vespa-cloud.ipynb:23]:
```suggestion
    "This notebook is intended to provide some clarity on the different authentication methods needed when interacting with Vespa Cloud for different purposes.\n"
```

File: docs/sphinx/source/authenticating-to-vespa-cloud.ipynb [hunk 2]
```diff
@@ -0,0 +1,984 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4f8c1448",
+   "metadata": {},
+   "source": [
+    "<div class=\"alert alert-info\">\n",
+    "    Refer to <a href=\"https://pyvespa.readthedocs.io/en/latest/troubleshooting.html\">troubleshooting</a>\n",
```
Comment (kkraune) [inline docs/sphinx/source/authenticating-to-vespa-cloud.ipynb:32]: troubleshooting link did not work inside the alert (when testing locally)

File: docs/sphinx/source/authenticating-to-vespa-cloud.ipynb [hunk 3]
```diff
@@ -0,0 +1,984 @@
...
+    "| Delete data                                                                                  | ❌            | ✅         |                                                                                            |\n",
+    "| [Visiting](https://docs.vespa.ai/en/visiting.html)                                           | ❌            | ✅         |                                                                                            |\n",
+    "| [Monitoring](https://cloud.vespa.ai/en/monitoring)                                           | ❌            | ✅         |                                                                                            |\n",
+    "| Get application package                                                                      | ✅            | ❌         |                                                                                            |\n",
+    "| [`vespa auth login`](https://docs.vespa.ai/en/reference/vespa-cli/vespa_auth_login.html)     | ✅            | ❌         | Interactive control-plane login in browser                                                 |\n",
```
Comment (kkraune) [inline docs/sphinx/source/authenticating-to-vespa-cloud.ipynb:126]: these links did not display well for me, maybe the local processing ...

File: docs/sphinx/source/authenticating-to-vespa-cloud.ipynb [hunk 4]
```diff
@@ -0,0 +1,984 @@
...
+   "id": "bfc6b343",
+   "metadata": {},
+   "source": [
+    "## Defining your application package\n",
+    "\n",
+    "An [application package](https://docs.vespa.ai/en/application-packages.html) is the whole Vespa application configuration.\n",
+    "It can either be constructed directly from python (as we will do below) or initalized from a path, for example by cloning a sample application from the [Vespa sample apps](https://github.com/vespa-engine/sample-apps).\n",
+    "\n",
+    "<div class=\"alert alert-info\">\n",
+    "    Tip: You can use the command <a href=\"https://docs.vespa.ai/en/reference/vespa-cli/vespa_clone.html#examples\">vespa clone album-recommendation my-app</a> to clone a single sample app if you have the Vespa CLI installed.\n",
```
Comment (kkraune) [inline docs/sphinx/source/authenticating-to-vespa-cloud.ipynb:168]: not sure why links do not show in alerts...

File: docs/sphinx/source/authenticating-to-vespa-cloud.ipynb [hunk 5]
```diff
@@ -0,0 +1,984 @@
...
+  {
+   "cell_type": "markdown",
+   "id": "197c0a27",
+   "metadata": {},
+   "source": [
+    "The following will upload the application package to Vespa Cloud Dev Zone (`aws-us-east-1c`), read more about [Vespa Zones](https://cloud.vespa.ai/en/reference/zones.html).\n",
+    "The Vespa Cloud Dev Zone is considered as a sandbox environment where resources are down-scaled and idle deployments are expired automatically.\n",
+    "For information about production deployments, see the following [docs](https://cloud.vespa.ai/en/reference/deployment).\n",
+    "\n",
+    "> Note: Deployments to dev and perf expire after 7 days of inactivity, i.e., 7 days after running deploy. This applies to all plans, not only the Free Trial. Use the Vespa Console to extend the expiry period, or redeploy the application to add 7 more days.\n"
```
Comment (kkraune) [inline docs/sphinx/source/authenticating-to-vespa-cloud.ipynb:336]: I think this is 14 days now

Comment (kkraune) [reply docs/sphinx/source/authenticating-to-vespa-cloud.ipynb:336]: ... until credits is launched ... :-)

**Review by kkraune — APPROVED** [review]

File: docs/sphinx/source/authenticating-to-vespa-cloud.ipynb (hunk 5 above)
Comment (thomasht86) [reply docs/sphinx/source/authenticating-to-vespa-cloud.ipynb]: Changed it now. Will keep a note to self to change when credits is launched.

File: docs/sphinx/source/authenticating-to-vespa-cloud.ipynb (hunk 3 above)
Comment (thomasht86) [reply docs/sphinx/source/authenticating-to-vespa-cloud.ipynb]: Tried to build HTML, and indeed, they did not display well. 
Fixed by removing the `pre`-tags

**Review by kkraune — APPROVED** [review]
👍

**Pull request merged**